checkov/terraform_json/parser.py

```diff
@@ -0,0 +1,199 @@
+                        template = template[0]
+            except Exception:
+                logger.error(f"Template {file_path} is malformed: {err.problem}")
+                logger.error(f"Tried to parse {file_path} as JSON", exc_info=True)
+    except YAMLError:
+        pass
+
+    if template is None or template_lines is None:
+        return None, None
+
+    return template, template_lines
+
+
+def load(file_path: Path) -> tuple[dict[str, Any], list[tuple[int, str]]]:
+    """Load the given JSON file"""
+
+    try:
+        content = file_path.read_text()
+    except UnicodeDecodeError:
+        logging.debug(f"Encoding for file {file_path} is not UTF-8, trying to detect it")
+        content = str(from_path(file_path).best())
+
+    if not all(key in content for key in ("resource", "provider")):
+        return {}, []
+
+    file_lines = [(idx + 1, line) for idx, line in enumerate(content.splitlines(keepends=True))]
+
+    template: "dict[str, Any] | list[dict[str, Any]]" = loader.loads(content=content)
+    if not template:
+        template = {}
+    if isinstance(template, list):
+        template = template[0]
+
+    if template:
+        template = prepare_definition(template)
+
+    return template, file_lines
+
+
+def prepare_definition(definition: dict[str, Any]) -> dict[str, Any]:
+    definition_new: dict[str, Any] = {}
+
+    for block_type, blocks in definition.items():
+        if block_type == COMMENT_FIELD_NAME or block_type in LINE_FIELD_NAMES:
+            continue
+
+        definition_new[block_type] = []
+        for block_name, config in blocks.items():
+            if block_name == COMMENT_FIELD_NAME or block_name in LINE_FIELD_NAMES:
+                continue
+
+            if block_type == BlockType.RESOURCE:
+                # resource have an extra nested level resource_type -> resource_name -> resource_config
+                for resource_name, resource_config in config.items():
+                    if resource_name in IGNORE_FILED_NAMES:
+                        continue
+                    definition_new[block_type].append({block_name: {resource_name: hclify(obj=resource_config)}})
+            elif block_type == BlockType.PROVIDER:
+                # provider are stored as a list, which we need to move one level higher to add the name
+                for provider_config in config:
+                    definition_new[block_type].append({block_name: hclify(obj=provider_config)})
+            else:
+                definition_new[block_type].append({block_name: hclify(obj=config)})
+
+    return definition_new
+
+
+def hclify(
+    obj: dict[str, Any],
+    conf: dict[str, Any] | None = None,
+    parent_key: str | None = None,
+) -> dict[str, list[Any]]:
+    ret_dict = {}
+
+    if not isinstance(obj, dict):
+        raise Exception("this method receives only dicts")
+
+    for key, value in obj.items():
+        if key == COMMENT_FIELD_NAME:
+            # don't hclify the comment block
```

Comment:
did you just coin a new word? 🤣

Reply:
sounds like 😄 but no, we use the name in the Terraform plan parser